two_factor_otp_auth/controllers/main.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2019 VentorTech OU
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0).
+
+import logging
+
+from odoo import http, _
+from odoo.addons.web.controllers.main import Home
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from ..exceptions import MissingOtpError, InvalidOtpError
+
+except IOError as error:
+    _logger.debug(error)
+
+
+class Login2fa(Home):
+
+    @http.route()
+    def web_login(self, redirect=None, **kw):
+        """
+        Overload core method to start Second Factor validation step
+        """
+        try:
+            response = super(Login2fa, self).web_login(redirect, **kw)
+        except MissingOtpError:
+            # user will coming here while login process is not full successful
+            # come here after success validation of other credentials
+            # to start Second Factor (OTP) validation step
```

Comment:
Fix some typos in comment:

# user will get into this block if login process is not fully successful
# For example, when first login was successful, but 2FA token is missing
# So we can start second authentication step (OTP)

Reply:
done